examples/variational_autoencoder.py

```diff
@@ -41,13 +42,34 @@ def sampling(args):
 x_decoded_mean = decoder_mean(h_decoded)
 
 
-def vae_loss(x, x_decoded_mean):
-    xent_loss = original_dim * metrics.binary_crossentropy(x, x_decoded_mean)
-    kl_loss = - 0.5 * K.sum(1 + z_log_var - K.square(z_mean) - K.exp(z_log_var), axis=-1)
-    return xent_loss + kl_loss
+# placeholder loss
+def zero_loss(y_true, y_pred):
+    return K.zeros_like(y_pred)
+
+
+# Custom loss layer
+class CustomVariationalLayer(Layer):
+    def __init__(self, **kwargs):
+        self.is_placeholder = True
+        super(CustomVariationalLayer, self).__init__(**kwargs)
+
+    def vae_loss(self, x, x_decoded_mean):
+        xent_loss = original_dim * metrics.binary_crossentropy(x, x_decoded_mean)
+        kl_loss = - 0.5 * K.sum(1 + z_log_var - K.square(z_mean) - K.exp(z_log_var), axis=-1)
+        return K.mean(xent_loss + kl_loss)
+
+    def call(self, inputs):
+        x = inputs[0]
+        x_decoded_mean = inputs[1]
+        loss = self.vae_loss(x, x_decoded_mean)
+        self.add_loss(loss, inputs=inputs)
+        # we don't use this output, but it has to have the correct shape:
+        return K.ones_like(x)
+
+loss_layer = CustomVariationalLayer()([x, x_decoded_mean])
+vae = Model(x, [loss_layer])
+vae.compile(optimizer='rmsprop', loss=[zero_loss])
```

Comment:
This is the error that one would except. If there is not loss on the output, why would you pass any target data?